diary/views.py

```diff
@@ -260,3 +262,111 @@ def get(self, request):
             response_status=status.HTTP_200_OK
         )
+
+
+class KeywordImgSaveView(APIView):
+    @transaction.atomic
+    @swagger_auto_schema(operation_description="키워드별 이미지 저장", request_body=KeyWordImgSaveRequest, responses={"201": '저장 성공'})
+    def post(self, request):
+        requestSerial = KeyWordImgSaveRequest(data=request.data)
+
+        isValid, response_status = requestSerial.is_valid()
+        # 유효성 검사 통과하지 못한 경우
+        if not isValid:
+            return ApiResponse.on_fail(requestSerial.errors, response_status=response_status)
+
+        request = requestSerial.validated_data
+
+        # keyword 객체 가져오기
+        keyword = Keywords.objects.get(id=request.get('keywordId'))
+
+        # imgUrl 저장
+        keyword.imgUrl = request.get('imgUrl')
+        keyword.save()
+
+        return ApiResponse.on_success(
+            result=KeywordSerializer(keyword).data,
+            response_status=status.HTTP_201_CREATED
+        )
+
+
+class DiaryImgSaveView(APIView):
+    @transaction.atomic
+    @swagger_auto_schema(operation_description="일기 이미지 저장", request_body=DiaryImgSaveRequest, responses={"201": '저장 성공'})
+    def post(self, request):
+        requestSerial = DiaryImgSaveRequest(data=request.data)
+
+        isValid, response_status = requestSerial.is_valid()
+        # 유효성 검사 통과하지 못한 경우
+        if not isValid:
+            return ApiResponse.on_fail(requestSerial.errors, response_status=response_status)
+
+        request = requestSerial.validated_data
+
+        # keyword 객체 가져오기
+        diary = Diary.objects.get(id=request.get('diaryId'))
+
+        # imgUrl 저장
+        diary.imgUrl = request.get('imgUrl')
+        diary.save()
+
+        return ApiResponse.on_success(
+            result=DiaryResultResponse(diary).data,
+            response_status=status.HTTP_201_CREATED
+        )
+    
+
+class KeywordImgPagingView(APIView):
+    @transaction.atomic
+    @swagger_auto_schema(operation_description="키워드별 사진 페이징", query_serializer=FindKeywordImgRequest)
+    def get(self, request):
+        requestSerial = FindKeywordImgRequest(data=request.query_params)
+
+        isValid, response_status = requestSerial.is_valid()
+        # 유효성 검사 통과하지 못한 경우
+        if not isValid:
+            return ApiResponse.on_fail(requestSerial.errors, response_status=response_status)
+
+        request = requestSerial.validated_data
+
+        keyword_objects = Keywords.objects.filter(keyword=request.get('keyword'))
+
+        # 필터링된 객체가 없을 경우 404 반환
+        if not keyword_objects:
+            return ApiResponse.on_fail({"message": "아직 그려진 그림이 없습니다."}, status.HTTP_404_NOT_FOUND)
+        
+        # 최신순으로 정렬
+        keyword_objects = keyword_objects.order_by('-updated_at')
+
+        # imgUrl 필드만 가져와서 리스트로 변환
+        keyword_img_urls = list(keyword_objects.values_list('imgUrl', flat=True))
+        page = request.get('page')
+        pageSize = request.get('pageSize')
+
+        paginator = Paginator(keyword_img_urls, pageSize)
```

Comment:
그리고 여기 부분 쿼리셋을 리스트로 변환한다음 페이지 네이션 하는데

이러면 성능이슈가 있을수 있을것 같아 리스트 변환말고 쿼리셋을 직접 페이지네이션 하는걸로 ㄱㄱ

```suggestion

        # imgUrl 필드만 가져와서 리스트로 변환
        keyword_img_urls = keyword_objects.values_list('imgUrl', flat=True)
        page = request.get('page')
        pageSize = request.get('pageSize')

        paginator = Paginator(keyword_img_urls, pageSize)
```